wxmaxima out of date

I tried to update `wxmaxima` myself, but simply bumping the version number leads to the fallowing error:

```
-- Looking for wxWidgets in:           /nix/store/fnz241dg38svrdgqjxfxv8d68izbf719-wxwidgets-3.2.1
-- Looking for wxWidgets libraries in: 
CMake Error at /nix/store/blqlkgmvs12h8m8rhbv7pmyj7jyxxp37-cmake-3.24.3/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find wxWidgets (missing: wxWidgets_LIBRARIES) (Required is at
  least version "3")
Call Stack (most recent call first):
  /nix/store/blqlkgmvs12h8m8rhbv7pmyj7jyxxp37-cmake-3.24.3/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /nix/store/blqlkgmvs12h8m8rhbv7pmyj7jyxxp37-cmake-3.24.3/share/cmake-3.24/Modules/FindwxWidgets.cmake:977 (find_package_handle_standard_args)
  CMakeLists.txt:137 (find_package)


-- Configuring incomplete, errors occurred!
```

###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: [wxmaxima](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/science/math/wxmaxima/default.nix)
current version: [22.05](https://github.com/wxMaxima-developers/wxmaxima/releases/tag/Version-22.05.0) 
desired version: [22.12](https://github.com/wxMaxima-developers/wxmaxima/releases/tag/Version-22.12.0)

###### Notify maintainers
maintainers: @doronbehar 

###### Note for maintainers

Please tag this issue in your PR.

--- a/pkgs/applications/science/math/wxmaxima/default.nix
+++ b/pkgs/applications/science/math/wxmaxima/default.nix
@@ -7,17 +7,18 @@
 , maxima
 , wxGTK
 , gnome
+, glib
 }:
 
 stdenv.mkDerivation rec {
   pname = "wxmaxima";
-  version = "22.05.0";
+  version = "22.12.0";
 
   src = fetchFromGitHub {
     owner = "wxMaxima-developers";
     repo = "wxmaxima";
     rev = "Version-${version}";
-    sha256 = "sha256-pcKnEjJmvMXCBpjtOSLyl4I0x3fjh0os9Sdp39I2Re0=";
+    sha256 = "sha256-RT6y4M6LQD1fXJcjtdSXnDmoJvv160g2asdV4WtTcok=";
   };
 
   buildInputs = [
@@ -25,12 +26,18 @@ stdenv.mkDerivation rec {
     maxima
     # So it won't embed svg files into headers.
     gnome.adwaita-icon-theme
+    # So it won't crash under Sway.
+    glib
   ];
 
   nativeBuildInputs = [
     wrapGAppsHook
     cmake
     gettext
+  ];
+
+  cmakeFlags = [
+    "-DwxWidgets_LIBRARIES=${wxGTK}/lib"
   ];
 
   preConfigure = ''

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -36556,7 +36556,9 @@ with pkgs;
   };
 
   wxmaxima = callPackage ../applications/science/math/wxmaxima {
-    wxGTK = wxGTK32;
+    wxGTK = wxGTK32.override {
+      withWebKit = true;
+    };
   };
 
   pari = callPackage ../applications/science/math/pari { tex = texlive.combined.scheme-basic; };

